pkg/cli/cli.go

```diff
@@ -1362,27 +1362,55 @@ func runConnectionWebSocketAgentCommand(flags *ConnectFlags, c *command, args []
 		return err
 	}
 	adbPort := l.Addr().(*net.TCPAddr).Port
-	go func() {
-		if err := opts.ADBServerProxy.Connect(adbPort); err != nil {
+	adbErrCh := make(chan error)
+	go func(adbErrCh chan error) {
+		err := opts.ADBServerProxy.Connect(adbPort)
+		if err != nil {
 			c.PrintErrf("Failed to connect ADB to device: %v\n", err)
 		}
-	}()
+		adbErrCh <- err
+	}(adbErrCh)
 
-	tcpListener, ok := l.(*net.TCPListener)
-	if !ok {
-		c.PrintErrf("Listener is not a TCPListener\n")
-		return errors.New("Listener is not a TCPListener")
-	}
-	tcpListener.SetDeadline(time.Now().Add(time.Duration(adbListenTimeoutSec) * time.Second))
+	listenResultCh := make(chan net.Conn)
+	listenErrCh := make(chan error)
```

Comment:
nit: No need for two channels if you create a small struct to hold the two values:
```
type listenResult struct {
    conn net.Conn
    err  error
}
listenCh := make(chan, listenResult)
```

Reply:
Merged separated channels into one channel.